OCD-3985: Re-enable Developer Attestation Reports

kekey1 opened the pull request.

kekey1 committed:
refactor: wip: Use Change Request CSV Report structre for Pending CR Rpt

[#OCD-3985]

kekey1 committed:
Merge branch 'staging' into OCD-3985

kekey1 committed:
refactor: Use updated CR report structure for pending CR report

The Pending Change Request report is updated to use the two-step
process of 1) search, and 2) get search result details in a
multi-threaded way. The presentation code was moved out to a Presenter
class.

[#OCD-3985]

kekey1 committed:
wip: Make the Developer Attestation report work with attestation form

[#OCD-3985]

kekey1 committed:
Merge branch 'staging' into OCD-3985

kekey1 committed:
feat: Re-enable the Developer Attestation Check-In Report

There was common functionality being used in the check-in report and the
regular Developer Attestation Report - in fact, the check-in report was
calling the data collector for the attesation report which seemed like a
lot of extra work (running both reports essentially). I refactored the
shared functionality into the DeveloperAttestationPeriodCalculator class.
More calculations could occur there but for now, there is just 1.

[#OCD-3985]

kekey1 committed:
fix: Optionally join the ACB Address in change request HQL

[#OCD-3985]

kekey1 committed:
refactor: Remove unused class variable

[#OCD-3985]

kekey1 committed:
feat: Use current calendar year for RWT validation in DevAtt report

[#OCD-3985]

kekey1 committed:
Merge branch 'staging' into OCD-3985

kekey1 committed:
fix: Page through all listings for dev when determining RWT.

This commit also makes the Assurances criteria flag-based. Before
ERD-Phase-2 the criteria checked include g8 and g9-original. After the
flag is turned on those criteria are no longer included in the check.

[#OCD-3985]

kekey1 committed:
feat: Reduce logging of search requests in Developer Attestation Rpt

[#OCD-3985]

andlar commented on chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/changerequest/presenter/PendingChangeRequestPresenter.java:
```diff
@@ -0,0 +1,102 @@
+    }
+
+    protected boolean isSupported(ChangeRequest data) {
+        return true;
```
?

What is this?

kekey1 replied:
An implementation of an abstract method... the presenter can support some change requests (like just demographic ones, just attestation ones, or some other subset) or it can support them all, as the PendingChangeRequestPresenter does.

andlar commented on chpl/chpl-service/src/main/java/gov/healthit/chpl/service/realworldtesting/RealWorldTestingCriteriaService.java:
```diff
@@ -32,6 +32,7 @@ public RealWorldTestingCriteriaService(CertificationCriterionService certificati
 
     public List<CertificationCriterion> getEligibleCriteria(Integer year) {
         try {
+            System.out.println("Getting RWT criteria from year " + year);
```
Assuming this is debug code?

kekey1 replied:
Removed

kekey1 committed:
fix: Page through all developer listings for calculations

[#OCD-3985]

kekey1 committed:
refactore: Remove debug code

[#OCD-3985]

kekey1 committed:
Merge branch 'staging' into OCD-3985

kekey1 committed:
fix: Correctly order API and RWT attestation columns

[#OCD-3985]

kekey1 committed:
fix: Get attestation response from question ID instead of section ID

[#OCD-3985]

kekey1 committed:
fix: Untangle api criteria from assurances criteria variables

[#OCD-3985]

kekey1 committed:
Merge branch 'staging' into OCD-3985

kekey1 committed:
fix: Re-order API and RWT column data for check-in report

[#OCD-3985]

kekey1 committed:
Merge branch 'staging' into OCD-3985

kekey1 committed:
fix: Display ACBs separated with "; " instead of ","

[#OCD-3985]

kekey1 committed:
fix: Use correct query for check-in report

The Check-In Report was missing data for first-time attestation
submissions (i.e. not resubmissions) during any given
attestation period.

[#OCD-3985]

kekey1 committed:
fix: Only use published attestations for yes/no field of check-in rpt

[#OCD-3985]
